src/backend/distributed/operations/citus_tools.c

```diff
@@ -93,6 +101,122 @@ CheckConnectionToNode(char *nodeName, uint32 nodePort)
 }
 
 
+/*
+ * citus_check_cluster_node_health UDF performs connectivity checks from all the nodes to
+ * all the nodes, and report success status
+ */
+Datum
+citus_check_cluster_node_health(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	TupleDesc tupleDescriptor = NULL;
+	Tuplestorestate *tupleStore = SetupTuplestore(fcinfo, &tupleDescriptor);
+
+	StoreAllConnectivityChecks(tupleStore, tupleDescriptor);
+
+	/* clean up and return the tuplestore */
+	tuplestore_donestoring(tupleStore);
```

Comment:
minor: I think `tuplestore_donestoring` is not needed `#define tuplestore_donestoring(state) ((void) 0)`

Reply:
Apparently, PG do not need this call since 7.4 release.

What is really strange is that Tom Lane committed a change last year using this NOOP call https://github.com/postgres/postgres/commit/dd04e958c8b03c0f0512497651678c7816af3198

I will remove the call here, and maybe remove 20 other occurrences spread across 16 files on a separate PR.